### ISSUE TITLE
Ambient Temp Calibration

### ISSUE THREAD

**Issue by SeanPM5** [JeffSteinbok/hass-dreo#183]
Searched and couldn't find anything about this. I have a DR-HTF004S (Pilot Max S) and it has a feature where you can calibrate the temperature. It was originally reporting 83°F degrees (in the app and HA), but I've used the app feature to calibrate it down to -4°F degrees less. 

The Dreo app says 79°F (adjusted value), but HA sensor is still reporting 83°F after lengthy time and restart. It's seemingly not picking up the calibrated value. Is there any way to get it to use the adjusted temp?

![IMG_2333](https://github.com/user-attachments/assets/dfbcb5dd-4728-468e-8807-b5d2bfbc45dc)
![2F856E6A-CD96-45DE-81C2-7618FDDDE854](https://github.com/user-attachments/assets/06d37e21-fa17-4ad7-a445-8a73244f30e9)

[config_entry-dreo-01J4SESQHVM1DM0J5VF830S8QV.json](https://github.com/user-attachments/files/16883577/config_entry-dreo-01J4SESQHVM1DM0J5VF830S8QV.json)

**Comment by JeffSteinbok** [comment]
I don't see anything like that in the diagnostics (which have the full payload back from Dreo). It's possible it's just an app feature. If you delete the app and reinstall is it still there?

**Comment by SeanPM5** [comment]
Yes, just deleted and re-installed and it's still showing my adjusted / calibrated temp on the main home screen and controls screen. So I assume it's being stored server-side and not just in the app.

I do see this in the diagnostics, which seems to be the setting in question, but I don't see the calibrated temp value in the payload either.

                "preference": [
                  {
                    "id": "250",
                    "type": "Temperature Calibration",
                    "title": "device_control_temp_calibration",
                    "image": "ic_temp_cal"
                  }
                ],

**Comment by JeffSteinbok** [comment]
@zyonse any chance you have any idea?

**Comment by zyonse** [comment]
Could be this? Hard to say because my fan doesn't have that option in the app.

<img width="609" alt="Screenshot 2024-09-06 at 7 41 33 PM" src="https://github.com/user-attachments/assets/36ddf79d-dbd7-4ccf-9a5a-6ef4f694aeb7">
<img width="614" alt="Screenshot 2024-09-06 at 7 43 53 PM" src="https://github.com/user-attachments/assets/5451e8cb-cbd5-4ade-997b-1d67469f5414">

Funnily enough we actually have a config option in the homebridge plugin that does the exact same thing, setting an offset for the temp sensor.

**Comment by JeffSteinbok** [comment]
@zyonse does it not come back in the regular GetDeviceState call?

**Comment by JeffSteinbok** [comment]
Ok, just so I have it saved somewhere....this should be enough to sort this out and implement it.  
One thing we need to figure out is if we should allow setting it from here, or just factor it into the calculation of the actual temperature.  

I see a GET to:

`https://app-api-us.dreo-cloud.com/api/user-device/setting?timestamp=1725762806531&dataKey=kHafFanTempOffsetKey&deviceSn=REDACTED`

Which returns this:

`{"code":0,"msg":"OK","data":{"id":"1832608117463527425","deviceSn":"REDACTED","dataKey":"kHafFanTempOffsetKey","dataValue":"0"}}
`

I see a PUT to:

`https://app-api-us.dreo-cloud.com/api/user-device/setting?timestamp=1725762798003`

With this body to reduce the temperature 1 degree:

`{"dataKey":"kHafFanTempOffsetKey","dataValue":-1,"deviceSn":"REDACTED"}`

**Comment by JeffSteinbok** [comment]
Alright, @SeanPM5 can you try out the attached branch and see if it works for you?

**Comment by SeanPM5** [comment]
Tested the branch on a secondary HA instance:

- It picked up my -4 calibrated temp (74F) rather than 78F. Good!
- Then changed calibrated temp to -10 (from 78F to 68F) for testing purposes. Reloaded integration and it picked up that value too. Good!
- Lastly, I changed calibration to -1 in the Dreo app. This time, instead of reloading the HA integration I waited for the temperature state to update in HA by itself. It went from 68 to 69, so it was still using the -10 from above (last time I reloaded) rather than my more recent -1 app adjustment (that I didn't reload integration for).

So that is the only minor issue I found, if you adjust in the app the integration doesn't pick it up until reloaded. I consider that more of a minor edge case but just wanted to share my complete findings. Besides that one tiny thing, it's working great for me.
